BuySellStocks.java

```diff
@@ -0,0 +1,24 @@
+// Time Complexity : O(n) where n is the number of prices in the array
+// Space Complexity : O(1) 
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : Peak and Valley logic didnt strike but why it is giving "Guaranteed" solution? I was thinking along the lines of minimum Cost Price and max Selling Price.
```

Comment:
We will be doing this problem again towards the end. I will take this doubt of yours